sqlmesh/utils/dag.py

```diff
@@ -92,13 +92,28 @@ def sorted(self) -> t.List[T]:
             self._sorted = []
 
             unprocessed_nodes = self.graph
+            last_processed_nodes: t.Set[T] = set()
+
             while unprocessed_nodes:
                 next_nodes = {node for node, deps in unprocessed_nodes.items() if not deps}
 
                 if not next_nodes:
+                    cycle_candidates_msg = (
+                        "\nPossible model candidates to check for circular references: "
+                        + ", ".join(str(node) for node in unprocessed_nodes)
+                    )
+
+                    if last_processed_nodes:
+                        last_processed_msg = "\nLast models added to the DAG: " + ", ".join(
```

Comment:
Just a heads up, that the `DAG` implementation is pretty generic and may not just contain models. Should we reflect this fact in the error message and change `model` to eg. `node`?

Reply:
That makes sense, though it may make the error message a bit more cryptic because it might not be obvious that "nodes" are essentially models. I don't have a strong opinion on this though, happy to go either way.

Reply:
Btw I chatted with Chris and I'm looking into alternative metrics that are more helpful when troubleshooting circular DAGs. For example, display the last processed nodes & those nodes that had no dependencies removed in the last iteration.

Reply:
Replaced to use `node` instead of `model`